packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -1473,6 +1485,214 @@ class _SelectableFragment with Selectable, ChangeNotifier implements TextLayoutM
     return SelectionUtils.getResultBasedOnRect(_rect, localPosition);
   }
 
+  TextPosition _closestWordBoundary(
+    ({TextPosition wordStart, TextPosition wordEnd}) wordBoundary,
+    TextPosition position,
+  ) {
+    final int differenceA = (position.offset - wordBoundary.wordStart.offset).abs();
+    final int differenceB = (position.offset - wordBoundary.wordEnd.offset).abs();
+    return differenceA < differenceB ? wordBoundary.wordStart : wordBoundary.wordEnd;
+  }
+
+  TextPosition _updateSelectionStartEdgeByWord(
+    ({TextPosition wordStart, TextPosition wordEnd})? wordBoundary,
```

Comment:
Is it possible to define a type for this?

otherwise I think it would be more readable if you keep them separated.

Reply:
Looks like this is possible 

```
/// The start and end positions for a word.
typedef _WordBoundaryRecord = ({TextPosition wordStart, TextPosition wordEnd});
```